LanguageWeaverProvider/Studio/TranslationProviderLanguageDirection.cs

```diff
@@ -0,0 +1,291 @@
+		private readonly LanguagePair _languagePair;
+
+		private TranslationUnit _currentTranslationUnit;
+		private Window _batchTaskWindow;
+
+		public TranslationProviderLanguageDirection(ITranslationProvider translationProvider, ITranslationOptions translationOptions, LanguagePair languagePair, ITranslationProviderCredentialStore credentialStore)
+		{
+			TranslationProvider = translationProvider;
+			_translationOptions = translationOptions;
+			_languagePair = languagePair;
+			CredentialManager.SetCredentials(credentialStore, translationOptions);
+		}
+
+		public ITranslationProvider TranslationProvider { get; private set; }
+
+		public CultureCode SourceLanguage => _languagePair.SourceCulture;
+
+		public CultureCode TargetLanguage => _languagePair.TargetCulture;
+
+		public bool CanReverseLanguageDirection => false;
+
+		public ImportResult[] AddOrUpdateTranslationUnits(TranslationUnit[] translationUnits, int[] previousTranslationHashes, ImportSettings settings)
+		{
+			throw new NotImplementedException();
+		}
+
+		public ImportResult[] AddOrUpdateTranslationUnitsMasked(TranslationUnit[] translationUnits, int[] previousTranslationHashes, ImportSettings settings, bool[] mask)
+		{
+			throw new NotImplementedException();
+		}
+
+		public ImportResult AddTranslationUnit(TranslationUnit translationUnit, ImportSettings settings)
+		{
+			throw new NotImplementedException();
+		}
+
+		public ImportResult[] AddTranslationUnits(TranslationUnit[] translationUnits, ImportSettings settings)
+		{
+			throw new NotImplementedException();
+		}
+
+		public ImportResult[] AddTranslationUnitsMasked(TranslationUnit[] translationUnits, ImportSettings settings, bool[] mask)
+		{
+			throw new NotImplementedException();
+		}
+
+		public SearchResults SearchSegment(SearchSettings settings, Segment segment)
+		{
+			if (!_translationOptions.ProviderSettings.ResendDrafts && _currentTranslationUnit.ConfirmationLevel != ConfirmationLevel.Unspecified)
+			{
+				var targetSegment = new Segment(_languagePair.TargetCulture);
+				targetSegment.Add(PluginResources.TranslationDraftNotResent);
+
+				var sR = new SearchResults { SourceSegment = segment.Duplicate() };
+				sR.Add(CreateSearchResult(segment, targetSegment));
+				return sR;
+			}
+
+			var sourceSegment = _translationOptions.ProviderSettings.IncludeTags
+							  ? segment.Duplicate()
+							  : RemoveTagsOnSegment(segment);
+			var searchResults = new SearchResults { SourceSegment = sourceSegment };
+			var searchResult = TranslateSegment(segment, sourceSegment);
+			searchResults.Add(searchResult);
+			return searchResults;
+		}
+
+		private SearchResult TranslateSegment(Segment segment, Segment sourceSegment)
+		{
+			CredentialManager.ValidateToken(_translationOptions);
+			var xliff = CreateXliffFile(new Segment[] { sourceSegment });
+			var mappedPair = GetMappedPair();
+			var translation = CloudService.Translate(_translationOptions.AccessToken, mappedPair, xliff).Result;
```

Comment:
Depending the synchronization context this is running under, this could deadlock: https://blog.stephencleary.com/2012/07/dont-block-on-async-code.html - if this is a concern for you (it might not be), I guess your best bet is to use ConfigureAwait(false) whenever awaiting a Task down the call stack.